x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationReloaderTests.java

```diff
@@ -503,6 +506,27 @@ public void testPEMTrustReloadException() throws Exception {
         assertThat(sslService.sslContextHolder(config).sslContext(), sameInstance(context));
     }
 
+    /**
+     * Tests that the reloader doesn't throw an exception if a file is configured to be outside of the config/ directory.
+     * These errors are handled correctly by the
+     */
+    public void testReadingOutsideConfigDirectoryDoesntFail() throws Exception {
```

Comment:
Based on the exceptions that get caught in `startWatching`, do we need another method for testing `IOException`? I suggest we modify this method to be more generic to cover both scenarios and randomize the exception.